### PR TITLE
[POT]: Requirements clean up

### PR DESCRIPTION
**Pull request by AlexKoff88**
### Details:
 - Removed pandas from the mandatory requirements since it is used only in the experimental algorithms
 - Revised numpy versions to comply with pre-build packages available for Python 3.11. (for the future)

### CODE REVIEW

**Review by KodiaqQ — APPROVED** [review]

File: tools/pot/setup.py [hunk 1]
```diff
@@ -65,15 +65,15 @@ def get_version():
     sys.argv.remove('--install-dev-extras')
 
 INSTALL_REQUIRES = [
+    "numpy>=1.16.6,<=1.23.1; python_version < '3.11'",
+    "numpy>=1.23.2; python_version >= '3.11'",
```
Comment (rkazants) [inline tools/pot/setup.py:69]: It will not be aligned with other requirements files:(

File: tools/pot/setup.py [hunk 2]
```diff
@@ -65,15 +65,15 @@ def get_version():
     sys.argv.remove('--install-dev-extras')
 
 INSTALL_REQUIRES = [
+    "numpy>=1.16.6,<=1.23.1; python_version < '3.11'",
```
Comment (rkazants) [inline tools/pot/setup.py:68]: Does POT not support 1.24 NumPy? Can you please tell why?

Comment (AlexKoff88) [reply tools/pot/setup.py:68]: I probably didn't get your message. The problem here is that there is no rebuild numpy 1.24 package for python_version < '3.11' which can cause the issue on the user's side.

File: tools/pot/setup.py (hunk 1 above)
Comment (AlexKoff88) [reply tools/pot/setup.py:69]: Does it cause any issues or it is just about misalighment?

**Review by rkazants — CHANGES_REQUESTED** [review]
Agreed to change the PR.
1.23 is not available for Python 3.11 and it starts to build this version.

File: tools/pot/setup.py (hunk 1 above)
Comment (AlexKoff88) [reply tools/pot/setup.py:69]: Numpy packages for Python 3.11 are available starting from 1.23.2. You can check yourself here: https://numpy.org/devdocs/release/1.23.2-notes.html

Comment (AlexKoff88) [reply tools/pot/setup.py]: Agreed that it is not relevant anymore since the numpy upper bound has been removed from the package requirements.

**Review by rkazants — APPROVED** [review]

**Review by p-wysocki — APPROVED** [review]

**Comment by glenn-jocher** [comment]
@AlexKoff88 thanks for removing pandas! The pandas==1.3.5 pin is causing us a lot of problems in Ultralytics YOLOv8. This is still in place in 2022.3.0 and it's been a while since that release. Is there another release coming soon that incorporates this PR?

**Comment by AlexKoff88** [comment]
Hi @glenn-jocher. You can try a pre-release of "openvino-dev": https://pypi.org/project/openvino-dev/2023.0.0.dev20230407/. "openvino" package is also available.

**Pull request merged**